src/core/Dropdown/Dropdown/Dropdown.tsx

```diff
@@ -241,37 +177,60 @@ export class Dropdown extends Component<DropdownProps> {
         id={id}
         {...passProps}
       >
-        <HtmlLabel
-          id={labelId}
-          {...labelProps}
-          className={dropdownClassNames.label}
-        >
-          {labelMode === 'hidden' ? (
-            <VisuallyHidden>{labelText}</VisuallyHidden>
-          ) : (
-            <Paragraph {...labelTextProps}>{labelText}</Paragraph>
-          )}
-        </HtmlLabel>
-        <ListboxInput {...listboxInputProps}>
-          <ListboxButton {...passDropdownButtonProps}>
-            {listboxDisplayValue}
-          </ListboxButton>
-          {!!ListboxPopoverComponentReplace ? (
-            <ListboxPopoverComponentReplace {...passDropdownPopoverProps}>
-              {this.dropdownItems(children, passDropdownItemProps)}
-            </ListboxPopoverComponentReplace>
-          ) : (
+        <HtmlDiv>
+          <LabelText
+            id={labelId}
+            labelMode={labelMode}
+            as="label"
+            optionalText={optionalText}
+          >
+            {labelText}
+          </LabelText>
+          <ListboxInput {...listboxInputProps}>
+            <ListboxButton
+              {...dropdownButtonProps}
+              id={buttonId}
+              className={classnames(dropdownClassNames.button, {
+                [dropdownClassNames.disabled]: !!disabled,
+              })}
+              {...buttonAriaLabelledByOverride}
```

Comment:
This could be done using ariaConditional prop util with a single liner. I'd prefer to have it here since there are quite a few derived props above the template.

Reply:
Could it, since the parameter defining if the aria prop should be there is not the aria prop itself?

Reply:
```
{...getConditionalAriaProp('aria-labelledby', [
            !!selectedValue ? ariaLabelledByIds : undefined,
          ])}
```
or even
{...getConditionalAriaProp('aria-labelledby', [
            !!selectedValue ? ariaLabelledBy : undefined,
            !!selectedValue ? labelId : undefined,
          ])}

Reply:
Not quite a single liner, but now in use. At least it's now in a more logical place in the code.

Reply:
Seems weird that both; ListboxInput and ListboxButton have same aria-labelledby ids. But it does work with the screenreader when tested.

Reply:
Probably would work even with one of those removed. Can both button and input be focused when navigation with modifier + arrow keys?

Reply:
Both are needed. If the button doesn't have a labelledBy, the visualPlaceholder gets read instead of the label if a visualPlaceholder is used.